processing/src/main/java/io/druid/query/extraction/MapLookupExtractor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.common.StringUtils;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+@JsonTypeName("map")
+public class MapLookupExtractor implements LookupExtractor
+{
+  private final Map<String, String> map;
+  @JsonCreator
+  public MapLookupExtractor(
+      @JsonProperty("map") Map<String, String> map
+  ){
+    this.map = Preconditions.checkNotNull(map, "map");
+  }
+
+  @JsonProperty
+  public Map<String, String> getMap(){
+    return ImmutableMap.copyOf(map);
+  }
+
+  @Nullable
+  @Override
+  public String apply(@NotNull String val)
+  {
+    return map.get(val);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    try {
+      final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+      for(Map.Entry<String, String> entry : map.entrySet()){
+        final String key = entry.getKey();
+        final String val = entry.getValue();
+        if(!Strings.isNullOrEmpty(key)){
+          outputStream.write(StringUtils.toUtf8(key));
+        }
+        if(!Strings.isNullOrEmpty(val)){
+          outputStream.write(StringUtils.toUtf8(val));
+        }
+      }
+      return outputStream.toByteArray();
```

Comment:
This *could* be something like an md5 hash of the key/value pairs if this proves to be too large for a key.